include/revng/Yield/Support/GraphLayout/Graphs.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <unordered_map>
+
+#include "revng/ADT/GenericGraph.h"
+#include "revng/Yield/Support/GraphLayout/Traits.h"
+
+namespace yield::layout {
+
+namespace detail {
+
+template<typename DataType>
+struct InputNode : public DataType {
+  Size Size = { 0, 0 };
+
+  using DataType::DataType;
+  InputNode(const DataType &Another) : DataType(Another){};
+  InputNode(DataType &&Another) : DataType(Another){};
+};
+
+template<typename DataType>
+struct OutputNode : public DataType {
+  Point Center = { 0, 0 };
+  Size Size = { 0, 0 };
+
+  using DataType::DataType;
+  OutputNode(const DataType &Another) : DataType(Another){};
+  OutputNode(DataType &&Another) : DataType(Another){};
+};
+
+template<typename DataType>
+struct InputEdge : public DataType {
+  using DataType::DataType;
+  InputEdge(const DataType &Another) : DataType(Another){};
+  InputEdge(DataType &&Another) : DataType(Another){};
+};
+
+template<typename DataType>
+struct OutputEdge : public DataType {
```

Comment:
Same here. Shouldn't this inherit from `InputEdge`?

Reply:
Closing this in favour of [another thread](https://github.com/revng/revng/pull/278#discussion_r1146082077).